grouper/user_permissions.py

```diff
@@ -32,6 +39,8 @@ def user_has_permission(session, user, permission, argument=None):
     Returns:
         bool: Whether or not this user fulfills the permission.
     """
+    if not user:
```

Comment:
I wonder if making `user` optional has some subtle consequences somewhere, say by making a bad request do _something_ instead of erroring out...

Probably not, but could you note the behavior on no user input in the docstring? :)

Reply:
I can't remember why I added this and it's not necessary. Both mypy and the test suite are happy with making it non-optional. I'll revert that change before landing. Good catch!